### DISCUSSION TITLE
Having trouble with using MacPorts on new Macbook Air M2

### DISCUSSION THREAD

**Discussion by aliebman** [OpenModelica/OpenModelica#9810] [General]
Hi folks,
Trying to install everything from scratch as per the instructions on github here.

Gets stuck at this stage:

sudo port install cmake ccache qt5 qt5-qtwebkit autoconf boost OpenSceneGraph openjdk11

**Comment by mahge** [comment]
Hi 👋.
I think we need a little more info. Do you get any error messages? How are you stuck?

**Comment by aliebman** [comment]
Sorry, I forgot to attach!
Password:
--->  Computing dependencies for cmake
--->  Cleaning cmake
--->  Computing dependencies for ccache
--->  Cleaning ccache
--->  Computing dependencies for qt5
--->  Cleaning qt5
--->  Computing dependencies for qt5-qtwebkit
--->  Fetching archive for qt5-qtwebkit
--->  Attempting to fetch qt5-qtwebkit-5.9.2_4.darwin_22.arm64.tbz2 from https://packages.macports.org/qt5-qtwebkit
--->  Attempting to fetch qt5-qtwebkit-5.9.2_4.darwin_22.arm64.tbz2 from http://jog.id.packages.macports.org/macports/packages/qt5-qtwebkit
--->  Attempting to fetch qt5-qtwebkit-5.9.2_4.darwin_22.arm64.tbz2 from https://ywg.ca.packages.macports.org/mirror/macports/packages/qt5-qtwebkit
--->  Building qt5-qtwebkit
Error: Failed to build qt5-qtwebkit: command execution failed
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_aqua_qt5/qt5-qtwebkit/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port qt5-qtwebkit failed

**Comment by mahge** [comment]
We have never tried to build OpenModelica on M2 machines so there is probably not much we can tell you. This seems to be a macports problem guessing from the output. qt5-qtwebkit has been deprecated for a while so it is possible that macports will have issues building it on newer architecture.

That said, if you post the log file here, maybe we can make an educated guess. The relevant log file is
`/opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_aqua_qt5/qt5-qtwebkit/main.log`

**Comment by aliebman** [comment]
Thanks!
Here it is. It would be cool if we could sort it out.

Kind Regards,

ARIEL LIEBMAN (PhD)

Director
Monash Energy Institute (MuEI)
monash.edu/energy-institute

Professor of Sustainable Energy Systems
Department of Data Science & AI, Faculty of IT

Strategic Advisor
RACE for 2030 CRC

For appointments:
Please contact my assistant, Rhian ***@***.*** Mon-Wed and Fri.

Contact
M: +61 437 028 771

Monash University
20 Exhibition Walk
Clayton, VIC 3800, Australia
CRICOS Provider: Monash University 00008C/01857J

We acknowledge and pay respects to the Elders and Traditional Owners of the land on which our four Australian campuses stand. Information for Indigenous Australians
We're committed to diversity and inclusion

**Comment by mahge** [comment]
There is nothing attached.

**Reply by aliebman** [reply]
Oops. Sorry. Don't know what happened there! Here it is. Thanks!
[main.log.zip](https://github.com/OpenModelica/OpenModelica/files/10166263/main.log.zip)

**Comment by aliebman** [comment]
Hi, were you able to download the log file I uploaded finally?

**Comment by mahge** [comment]
Hi, sorry for the delay. I forgot to update you. Unfortunately, there is really nothing we can do. This should probably be reported (if not reported yet) to the MacPorts issue tracker at https://trac.macports.org/newticket (you need to create an account). You can attach the log when you submit a ticket to them. However, there is a big chance they might not fix this since `qt-webkit` has been deprecated. 

On the other hand, we are also dropping support for `qt-webkit` and moving to `qt-webengine` soon. You can follow PR #9620 for updates regarding the migration. Hopefully it will happen soon.

**Comment by aliebman** [comment]
Thanks!
I’m happy to wait. I thought that might be the outcome.
Is that also why I can’t seem to run the nightly builds after installing them from here?

https://build.openmodelica.org/mac/pkg/arm64/nightly/

There used to be a Homebrew pathway. I that now not working on Mac either?

Thanks for all your help and in advance in case the above questions get me somewhere (or not 😀)


Kind Regards,

ARIEL LIEBMAN (PhD)

Director
Monash Energy Institute (MuEI)
monash.edu/energy-institute

Professor of Sustainable Energy Systems
Department of Data Science & AI, Faculty of IT

Strategic Advisor
RACE for 2030 CRC

For appointments:
Please contact my assistant, Rhian ***@***.*** Mon-Wed and Fri.

Contact
M: +61 437 028 771

Monash University
20 Exhibition Walk
Clayton, VIC 3800, Australia
CRICOS Provider: Monash University 00008C/01857J

We acknowledge and pay respects to the Elders and Traditional Owners of the land on which our four Australian campuses stand. Information for Indigenous Australians
We're committed to diversity and inclusion

**Reply by mahge** [reply]
> I’m happy to wait. I thought that might be the outcome. Is that also why I can’t seem to run the nightly builds after installing them from here? https://build.openmodelica.org/mac/pkg/arm64/nightly/ There used to be a Homebrew pathway. I that now not working on Mac either? Thanks for all your help and in advance in case the above questions get me somewhere (or not 😀)

Yes. To use the packages you need the dependencies installed, which, right now, includes qt-webkit. I am sorry but as far as I can see there is no way around it at the moment. Maybe something else out there provides binary packages for qt-webkit but I am not sure. 

> There used to be a Homebrew pathway. I that now not working on Mac either?

`hoembrew` will probably be the same story. As far as I know, `homebrew` has completely dropped support for qt-webkit. So it is worse than `macports` in that regard. 

> Thanks for all your help and in advance in case the above questions get me somewhere (or not 😀)

Sorry that we do not have good news for you yet. Keep an eye on #9620. We had plans to finish it up after the 1.20.0 release which just went out.

**Comment by aliebman** [comment]
Thanks!

All good.

I tried to do the build on my older x86 MacBook Pro. Build work but nothing runs 🤣 Any tips?

Kind Regards,

ARIEL LIEBMAN (PhD)

Director
Monash Energy Institute (MuEI)
monash.edu/energy-institute

Professor of Sustainable Energy Systems
Department of Data Science & AI, Faculty of IT

Strategic Advisor
RACE for 2030 CRC

For appointments:
Please contact my assistant, Rhian ***@***.*** Mon-Wed and Fri.

Contact
M: +61 437 028 771

Monash University
20 Exhibition Walk
Clayton, VIC 3800, Australia
CRICOS Provider: Monash University 00008C/01857J

We acknowledge and pay respects to the Elders and Traditional Owners of the land on which our four Australian campuses stand. Information for Indigenous Australians
We're committed to diversity and inclusion

**Comment by mahge** [comment]
If build works there is a big chance the whole thing will work. Do you get any error messages? How did you compile OpenModelica?

**Comment by aliebman** [comment]
Thanks. Same way it said on the instructions I used form the openmodelica website:

https://github.com/OpenModelica/OpenModelica/blob/master/README.cmake.md#33-macos


Kind Regards,

ARIEL LIEBMAN (PhD)

Director
Monash Energy Institute (MuEI)
monash.edu/energy-institute

Professor of Sustainable Energy Systems
Department of Data Science & AI, Faculty of IT

Strategic Advisor
RACE for 2030 CRC

For appointments:
Please contact my assistant, Rhian ***@***.*** Mon-Wed and Fri.

Contact
M: +61 437 028 771

Monash University
20 Exhibition Walk
Clayton, VIC 3800, Australia
CRICOS Provider: Monash University 00008C/01857J

We acknowledge and pay respects to the Elders and Traditional Owners of the land on which our four Australian campuses stand. Information for Indigenous Australians
We're committed to diversity and inclusion

**Comment by mahge** [comment]
You have to give us more information. There is no way we can help if we do not know what is happening. What do you mean by "nothing runs"? Do you get errors?

If you followed the instructions exactly, then there should be a folder named `OpenModelica/build_cmake/install_cmake/bin/omc`. what do you get if you go into this folder and run this in a command line:
```
omc --version
```
